app/src/main/java/org/dhis2/data/forms/dataentry/fields/image/ImageHolder.kt

```diff
@@ -0,0 +1,67 @@
+package org.dhis2.data.forms.dataentry.fields.image
+
+import android.view.View
+import androidx.databinding.ObservableField
+import io.reactivex.processors.FlowableProcessor
+import org.dhis2.Bindings.Bindings
+import org.dhis2.data.forms.dataentry.fields.FormViewHolder
+import org.dhis2.data.forms.dataentry.fields.RowAction
+import org.dhis2.databinding.FormImageBinding
+
+class ImageHolder(
+    private val imageBinding: FormImageBinding,
+    processor: FlowableProcessor<RowAction?>,
+    imageSelector: ObservableField<String?>
+) : FormViewHolder(imageBinding) {
+
+    private val currentSelector: ObservableField<String?> = imageSelector
+    private var isEditable = false
+    private var model: ImageViewModel? = null
+
+    init {
+        itemView.setOnClickListener {
+            if (isEditable) {
+                val label = model!!.optionDisplayName()
+                val code = model!!.optionCode()
```

Comment:
Will `model` never be null? if not, should we set "" to label and code??

Reply:
At this point in the code, the model won't be null and the optionDisplayName label and optionCode can't be null